Source/ORTS.Common/VirtualFileSystem.cs

```diff
@@ -0,0 +1,717 @@
+            {
+                case ZipArchiveEntry entry: return new StreamReader(entry.Open(), encoding);
+                case IArchiveEntry entry: return new StreamReader(entry.OpenEntryStream(), encoding);
+                case string path: return new StreamReader(path, encoding);
+                default: throw new FileNotFoundException($"VFS reading failed: {vfsPath}");
+            }
+        }
+
+        public static StreamReader OpenText(string vfsPath)
+        {
+            switch (PrepareForRead(vfsPath))
+            {
+                case ZipArchiveEntry entry: return new StreamReader(entry.Open(), Encoding.UTF8);
+                case IArchiveEntry entry: return new StreamReader(entry.OpenEntryStream(), Encoding.UTF8);
+                case string path: return File.OpenText(path);
+                default: throw new FileNotFoundException($"VFS reading failed: {vfsPath}");
+            }
+        }
+
+        public static Stream OpenRead(string vfsPath)
+        {
+            switch (PrepareForRead(vfsPath))
+            {
+                case ZipArchiveEntry entry: return entry.Open();
+                case IArchiveEntry entry: return entry.OpenEntryStream();
+                case string path: return File.OpenRead(path);
+                default: throw new FileNotFoundException($"VFS reading failed: {vfsPath}");
+            }
+        }
+
+        /// SharpCompress has the <see cref="Stream.Seek(long, SeekOrigin)"/> unimplemented,
+        /// so for reading a wav or dds within a VFS archive it needs to be loaded into the memory first.
+        public static Stream OpenReadWithSeek(string vfsPath)
+        {
+            var stream = OpenRead(vfsPath);
+            if (!stream.CanSeek)
+            {
+                var newStream = new MemoryStream();
+                stream.CopyTo(newStream);
+                stream.Close();
+                stream = newStream;
+                stream.Position = 0;
+            }
+            return stream;
+        }
+
+        public static DateTime GetLastWriteTime(string vfsPath)
+        {
+            switch (PrepareForRead(vfsPath))
+            {
+                case ZipArchiveEntry entry: return entry.LastWriteTime.DateTime;
+                case IArchiveEntry entry: return entry.LastModifiedTime ?? DateTime.MinValue;
+                case string path: return File.GetLastWriteTime(path);
+                default: throw new FileNotFoundException($"VFS reading failed: {vfsPath}");
+            }
+        }
+
+        public static Stream OpenCreate(string vfsPath)
+        {
+            switch (PrepareForWrite(vfsPath))
+            {
+                case Stream stream: return stream;
+                case string path: return new FileStream(path, FileMode.Create);
+                default: throw new DirectoryNotFoundException($"VFS writing failed: {vfsPath}");
+            }
+        }
+
+        public static void FileDelete(string vfsPath)
```

Comment:
Do not implement this. We never delete content files.

Reply:
We do. The generated *.or-binpat files need to be deleted, and they currently reside in the ROUTES\route\PATHS\ folder.

Reply:
Okay, that code is going to have to be fixed before we can proceed with VFS :(